Sources/SimplenoteFoundation/Extensions/String+Line.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+
+// MARK: - String + Line API(s)
+//
+extension String {
+
+    /// Returns a tuple with `(Range, Text)` of the Line at the specified location
+    ///
+    public func line(at index: String.Index) -> (Range<String.Index>, String) {
+        let range = rangeOfLine(at: index)
+        return (range, String(self[range]))
+    }
+
+    /// Returns the range of the line at the specified `String.Index`
+    ///
+    public func rangeOfLine(at index: String.Index) -> Range<String.Index> {
+        lineRange(for: index..<index)
```

Comment:
Not sure if we have a best practice for ranges. Should their be a space before and after the `..<` for readability? Thoughts?

Reply:
Normalizing, you're right, with a space things are much better!!